semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -39,9 +39,17 @@ type env = {
    * usually simple Instr, but can be also If when handling Conditional expr.
    *)
   stmts : stmt list ref;
+  (* When entering a loop, we create a two labels, one to jump to if a Continue stmt is found
+     and another to jump to if a Goto stmt is found. Since PHP supports breaking an arbitrary
+     number of loops up, we keep a stack of break labels instead of just one
+  *)
+  (* TODO: Bad to mix mutable + unmutable fields? *)
```

Comment:
I use mutable in environment only when info computed recursively needs to be used in other branches.
Here I don't think it's the case.